core/spec/lib/spree/event/adapters/default_spec.rb

```diff
@@ -243,6 +243,37 @@ def toggle
             expect(dummy2.run).to be(true)
           end
         end
+
+        describe '#with_listeners' do
+          it 'returns a new instance with given listeners' do
+            bus = described_class.new
+            dummy1, dummy2, dummy3 = Array.new(3) do
+              Class.new do
```

Comment:
@spaghetticode, if you see it (now you will ;)), I'll also use the counter class here as in #4203, but I'll rebase from master once that branch is merged.

Reply:
Done